### PR TITLE
string-functions: add instrunctions for LEFT() and LENGTH()

### PR DESCRIPTION
**Pull request by takaidohigasi**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Fix https://github.com/pingcap/docs/issues/15491

This PR adds instructions for the LEFT() and LENGTH function. (one of the [doc dash issues](https://github.com/orgs/pingcap/projects/51/views/3?pane=issue&itemId=45777998))

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by takaidohigasi** [comment]
[FYI] link for .md
https://github.com/takaidohigasi/docs/blob/750fc3ea92d8671b7f11660f366304f9aa6e2ba3/functions-and-operators/string-functions.md

**Review by dveeden — COMMENTED** [review]
The content looks very good. I think it explains the normal case as well as common corner cases.

I think the SQL statement and output should be in separate blocks like this:
````
```sql
SELECT 1
```
```
+---+
| 1 |
+---+
| 1 |
+---+
1 row in set (0.01 sec)
```
````

Maybe the text for `LENGTH()` should have a reference to `CHAR_LENGTH()`?

```
mysql> SELECT LENGTH('Café');
+-----------------+
| LENGTH('Café')  |
+-----------------+
|               5 |
+-----------------+
1 row in set (0.00 sec)

mysql> SELECT CHAR_LENGTH('Café');
+----------------------+
| CHAR_LENGTH('Café')  |
+----------------------+
|                    4 |
+----------------------+
1 row in set (0.00 sec)
```

There might be some more wording and formatting things that need to be done.

File: functions-and-operators/string-functions.md [hunk 1]
```diff
@@ -242,6 +242,7 @@ SELECT LEFT(NULL, 3);
 ### [`LENGTH()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_length)
 
 Return the length of a string in bytes.
+`LENGTH()` count a multibyte character as multi bytes while `CHAR_LENGTH()` count it as a single code point
```
Comment (dveeden) [inline functions-and-operators/string-functions.md:245]: Some things in Unicode make this quite complicated, for example flags.  However to me it looks like the text here is fully correct :smile: 


```
mysql> WITH n AS (SELECT 'abc' p UNION ALL SELECT 'Café' UNION ALL SELECT '😀' UNION ALL SELECT '🇪🇺') SELECT p, CHAR_LENGTH(p), LENGTH(p), HEX(p) FROM n;
+----------+----------------+-----------+------------------+
| p        | CHAR_LENGTH(p) | LENGTH(p) | HEX(p)           |
+----------+----------------+-----------+------------------+
| abc      |              3 |         3 | 616263           |                     
| Café     |              4 |         5 | 436166C3A9       |
| 🇪🇺         |              2 |         8 | F09F87AAF09F87BA |
| 😀         |              1 |         4 | F09F9880         |
+----------+----------------+-----------+------------------+
4 rows in set (0.01 sec)
```

**Review by dveeden — APPROVED** [review]

File: functions-and-operators/string-functions.md (hunk 1 above)
Comment (dveeden) [reply functions-and-operators/string-functions.md:245]: Note that the linter doesn't like the manual line break:

```
[2024-01-10T07:23:47.103Z] functions-and-operators/string-functions.md: this file has manual line breaks in the following lines:
[2024-01-10T07:23:47.103Z] 
[2024-01-10T07:23:47.103Z] MANUAL LINE BREAKS: L245
```

Maybe something like this, or merge it with line 244.

```suggestion

`LENGTH()` count a multibyte character as multi bytes while `CHAR_LENGTH()` count it as a single code point
```

Comment (takaidohigasi) [reply functions-and-operators/string-functions.md:245]: thanks for the review!

I agree with you and collation is more difficult for some languages related to the it...

File: functions-and-operators/string-functions.md [hunk 2]
```diff
@@ -173,10 +173,114 @@ Synonym for `LOWER()`.
 
 Return the leftmost number of characters as specified.
 
+* usage: LEFT(`str`, `len`)
+    * len: length of character to return
+    * str: original string to extract characters
+* if `len` is equal or less than 0, the function returns empty string
+* function is multibyte safe.
+* If any argument is `NULL`, the function returns `NULL`.
+
```
Comment (SeaRise) [inline functions-and-operators/string-functions.md:182]:
```suggestion
* If `len` is greater than the length of `str`, the function returns `str`.

```

Comment (SeaRise) [reply functions-and-operators/string-functions.md:182]: How about adding an explanation for `SELECT LEFT('ABCED', 6);`?

Comment (takaidohigasi) [reply functions-and-operators/string-functions.md:182]: thank you for the suggestion!

**Review by SeaRise — APPROVED** [review]
lgtm

File: functions-and-operators/string-functions.md [hunk 3]
```diff
@@ -173,10 +173,115 @@ Synonym for `LOWER()`.
 
 Return the leftmost number of characters as specified.
 
+* usage: LEFT(`str`, `len`)
+    * len: length of character to return
+    * str: original string to extract characters
```
Comment (qiancai) [inline functions-and-operators/string-functions.md:178]:
```suggestion

Syntax

```sql
LEFT(`str`, `len`)
```

- `str`: the original string to extract characters.
- `len`: length of characters to be returned. 

```

File: functions-and-operators/string-functions.md [hunk 4]
```diff
@@ -173,10 +173,115 @@ Synonym for `LOWER()`.
...
++------------------------------+
+| NULL                         |
++------------------------------+
+```
+
 ### [`LENGTH()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_length)
 
 Return the length of a string in bytes.
 
+`LENGTH()` count a multibyte character as multi bytes while `CHAR_LENGTH()` count it as a single code point
```
Comment (qiancai) [inline functions-and-operators/string-functions.md:247]:
```suggestion
`LENGTH()` counts a multibyte character as multi bytes while `CHAR_LENGTH()` counts a multibyte character as a single code point.
```

File: functions-and-operators/string-functions.md [hunk 5]
```diff
@@ -173,10 +173,115 @@ Synonym for `LOWER()`.
...
+
+* If any argument is `NULL`, the function returns `NULL`.
```
Comment (qiancai) [inline functions-and-operators/string-functions.md:249]:
```suggestion
If the argument is `NULL`, the function returns `NULL`.
```

File: functions-and-operators/string-functions.md
Comment (qiancai) [inline functions-and-operators/string-functions.md:1]: @takaidohigasi, thanks for your valuable contribution.  I've added some comments on this PR. PTAL.

File: functions-and-operators/string-functions.md [hunk 6]
```diff
@@ -173,10 +173,122 @@ Synonym for `LOWER()`.
...
+LEFT(`str`, `len`)
+```
+
+- `str`: the original string to extract characters.
+- `len`: the length of characters to be returned. 
+
+- If `len` is equal to or less than 0, the function returns an empty string.
+- If `len` is greater than the length of `str`, the function returns the length of `str`. 
+- If `str` contains a multibyte character, the function counts it as a single code point. 
+- If any argument is `NULL`, the function returns `NULL`.
```
Comment (qiancai) [inline functions-and-operators/string-functions.md:189]:
```suggestion
- `str`: the original string to extract characters. If `str` contains a multibyte character, the function counts it as a single code point. 
- `len`: the length of characters to be returned. 
    - If `len` is equal to or less than 0, the function returns an empty string.
    - If `len` is greater than the length of `str`, the function returns the length of `str`. 
- If any argument is `NULL`, the function returns `NULL`.
```

File: functions-and-operators/string-functions.md [hunk 7]
```diff
@@ -173,10 +173,120 @@ Synonym for `LOWER()`.
 
 Return the leftmost number of characters as specified.
```
Comment (qiancai) [inline functions-and-operators/string-functions.md:174]:
```suggestion
The `LEFT()` function returns a specified number of characters from the left side of a string.
```

File: functions-and-operators/string-functions.md [hunk 8]
```diff
@@ -173,10 +173,120 @@ Synonym for `LOWER()`.
...
++------------------------------+
+| LEFT(NULL, 3)                |
++------------------------------+
+| NULL                         |
++------------------------------+
+```
+
 ### [`LENGTH()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_length)
 
 Return the length of a string in bytes.
```
Comment (qiancai) [inline functions-and-operators/string-functions.md:250]:
```suggestion
The `LENGTH()` function returns the length of a string in bytes.
```

File: functions-and-operators/string-functions.md [hunk 9]
```diff
@@ -171,11 +171,121 @@ Synonym for `LOWER()`.
...
+Syntax:
+
+```sql
+LEFT(`str`, `len`)
+```
+
+- `str`: the original string to extract characters. If `str` contains a multibyte character, the function counts it as a single code point. 
+- `len`: the length of characters to be returned. 
+    - If `len` is equal to or less than 0, the function returns an empty string.
+    - If `len` is greater than the length of `str`, the function returns the length of `str`.
```
Comment (qiancai) [inline functions-and-operators/string-functions.md:186]:
```suggestion
    - If `len` is greater than the length of `str`, the function returns the original `str`. 
```

File: functions-and-operators/string-functions.md [hunk 10]
```diff
@@ -171,11 +171,121 @@ Synonym for `LOWER()`.
 
 ### [`LEFT()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_left)
 
-Return the leftmost number of characters as specified.
+The `LEFT()` function returns a specified number of characters from the left side of a string.
+
```
Comment (qiancai) [inline functions-and-operators/string-functions.md:175]:
```suggestion
```

File: functions-and-operators/string-functions.md [hunk 11]
```diff
@@ -171,11 +171,120 @@ Synonym for `LOWER()`.
...
+Syntax:
+
+```sql
+LEFT(`str`, `len`)
+```
+
+- `str`: the original string to extract characters. If `str` contains a multibyte character, the function counts it as a single code point. 
+- `len`: the length of characters to be returned. 
+    - If `len` is equal to or less than 0, the function returns an empty string.
+    - If `len` is greater than the length of `str`, the function returns the original `str`.
```
Comment (hfxsd) [inline functions-and-operators/string-functions.md:185]:
```suggestion
    - If `len` is equal to or greater than the length of `str`, the function returns the original `str`. 
```

**Review by hfxsd — APPROVED** [review]

**Review by qiancai — APPROVED** [review]

**Pull request merged**